src/pip/_internal/utils/subprocess.py

```diff
@@ -207,7 +206,7 @@ def call_subprocess(
         # Show the line immediately.
         log_subprocess(line)
         # Update the spinner.
-        if use_spinner:
+        if use_spinner and spinner:
```

Comment:
I'd rather `assert spinner` since `use_spinner` implies `spinner is not  None`.

Reply:
Thanks for the review. I have added the assert in https://github.com/pypa/pip/pull/8322/commits/4d55a57882972198ec5326138948af460359a2a4 .